packages/wrangler/src/config/validation.ts

```diff
@@ -327,7 +330,7 @@ function applyPythonConfig(
 	config: Config,
 	args: NormalizeAndValidateConfigArgs
 ) {
-	const mainModule = "script" in args ? args.script : config.main;
+	const mainModule = args.script ?? config.main;
```

Comment:
It is possible that the `args.script` property exists but is set to `undefined`, which means that we should still defer to `config.main` in that case.